src/solve.jl

```diff
@@ -323,23 +334,34 @@ function DiffEqBase.__init(
       =#
     else
       #if isadaptive(alg) || callback !== nothing
-        if alg_needs_extra_process(alg)
-          if alg===PL1WM()
-            m = length(rand_prototype)
-            if typeof(rand_prototype) <: Number
-              rand_prototype2 = nothing
-            else
-              rand_prototype2 = similar(rand_prototype,Int(m*(m-1)/2))
-              rand_prototype2 .= false
-            end
-            W = WienerProcess(t,rand_prototype,rand_prototype2,
+      if alg_needs_extra_process(alg)
+        if alg===PL1WM()
+          m = length(rand_prototype)
+          if typeof(rand_prototype) <: Number
+            rand_prototype2 = nothing
+          else
+            rand_prototype2 = similar(rand_prototype,Int(m*(m-1)/2))
+            rand_prototype2 .= false
+          end
+          W = WienerProcess(t,rand_prototype,rand_prototype2,
                              save_everystep=save_noise,
                              rng = Xorshifts.Xoroshiro128Plus(_seed))
+        elseif alg===RKMil_General()
+          m = length(rand_prototype)
+          if typeof(rand_prototype) <: Number || alg.p == nothing
+            rand_prototype2 = nothing
           else
```

Comment:
This whole thing is very messy and not general: it should definitely get a generalized form.